docs/architecture/adr-015-ibc-packet-receiver.md

```diff
@@ -0,0 +1,67 @@
+# ADR 015: IBC Packet Receiver
+
+## Changelog
+
+- 2019 Oct 22: Initial Draft
+
+## Context
+
+[ICS 26](https://github.com/cosmos/ics/tree/master/spec/ics-026-routing-module) defines function [`handlePacketRecv`](https://github.com/cosmos/ics/tree/master/spec/ics-026-routing-module#packet-relay). 
+`handlePacketRecv` executes per-module `onRecvPacket` callbacks, verifies the packet merkle proof, and pushes the acknowledgement bytes, if present,
+to the state. 
+
+The mechanism is similar to the transaction handling logic in `baseapp`. After authentication, the handler is executed, and 
+the authentication state change must be committed regardless of the result of the handler execution. 
+
+ICS 26 also requires acknowledgement writing which has to be done after the handler execution and also must be commited 
```

Comment:
Do we also need to write the acknowledgment as part of this ADR?

Reply:
If the acknowledgement is part of the return value of the handler it will need to be written by the code which calls the handler. This part is still unclear to me.

Reply:
We need a wrapper function inside the app handler, will add it